hare/broker.go

```diff
@@ -395,13 +395,13 @@ func (b *Broker) Register(ctx context.Context, id types.LayerID) (chan any, *Eli
 	if len(b.outbox) >= b.limit {
 		// unregister the earliest layer to make space for the new layer
 		// cannot call unregister here because unregister blocks and this would cause a deadlock
-		min := types.LayerID(0)
+		minimum := types.LayerID(0)
 		for lid := range b.outbox {
-			min = types.MinLayer(min, lid)
+			minimum = min(minimum, lid)
 		}
```

Comment:
Could it be replaced with `slices.Min`?

Reply:
Maybe with https://pkg.go.dev/slices#MinFunc. `types.MinLayer` returns the minimum non-zero LayerID, this might also be the cause of the failing tests.